test/test-utils.js

```diff
@@ -0,0 +1,61 @@
+"use strict";
+
+var JSZip = require("jszip");
+var fs = require("fs");
+var aps = require("azure-publish-settings");
+var path = require("path");
+var kuduApi = require("../");
+
+function ensureArtifacts(done) {
+    fs.mkdir("test/artifacts", function (err) {
+        if (err && err.code !== "EEXIST") {
+            return done(err);
+        }
+
+        done();
+    });
+}
+
+function createZipFile(localPath, files, cb) {
+    var generateOptions = {
+        type: "nodebuffer",
+        streamFiles: true
+    };
+
+    var zip = new JSZip();
+
+    Object.keys(files).forEach(function (key) {
+        zip.file(key, files[key]);
+    });
+
+    zip.generateNodeStream(generateOptions)
+        .pipe(fs.createWriteStream(localPath))
+        .on("error", cb)
+        .on("finish", cb);
+}
+
+function setupKudu(cb) {
+    return function (done) {
+        var settingsPath = process.env.TEST_PUBLISH_SETTINGS || path.join(__dirname, "test.PublishSettings");
```

Comment:
By convention, if we have a `test.PublishSettings` file in the tests directory it will be read and used to setup a new 'kudi-api' instance for testing. Alternatively a custom path could be set in an environment variable.

Could add a fallback behavior to use all environment variables, but I like the simplicity of the publish settings file. Can be downloaded and hooked up without setting up or even seeing any credentials!

![image](https://cloud.githubusercontent.com/assets/1434816/17834768/ca1681e8-6792-11e6-99e8-f499ee040d8c.png)